test/classes/nilability/question-mark-variations.chpl

```diff
@@ -51,7 +51,7 @@ proc factory1(type t) {
 proc factory2(type t) {
   return new t(1);
 }
-var d0 = factory1(C?);
+var d0 = factory1(C(?)?);
```

Comment:
@mppf: I'm catching up on this PR after falling behind last week.  This was the only case that I had local changes for that didn't end up in the PR.  What is it about the assertions in lines 33-36 that don't require / generate a warning for `C` without being `C(?)`?

Reply:
I only have a hypothesis here, so it's not something I know for sure. But I think that the reason is that the resolver is handling `==` on types before it gets to the part where it would generate a warning.

Arguably,  `==` on types is a reasonable function to allow generic types without `(?)` as an exception. It does not seem as likely to cause confusion if one of the actuals is generic, as compared to, say, a function accepting a `type T` that goes on to do `new T`. However, we can add additional warnings without disrupting language stability.